File: src/js/results.js
"use strict";
class BaseResult {
    constructor(success) {
        this.success = success;
    }
}

module.exports = {
    ParseSideResult: class ParseSideResult extends BaseResult {
        constructor(value) {
            let isNumber = typeof value === "number";
            super(isNumber);
            if (isNumber) {
                this.side = value;
            } else {
                this.errorMessage = value;
            }
        }
    },

    MatrixResult: class MatrixResult extends BaseResult {
        constructor(matrix, sum, side, success) {
            super(success);
            this.matrix = matrix;
            this.sum = sum;
            this.side = side;
            this.success = success;
        }
    }
}

File: src/py/results.py
class ParseSideResult:
    def __init__(self, value) -> None:
        if isinstance(value, int):
            self.success = True
            self.side = value
        else:
            self.success = False
            self.errorMessage = value

class MatrixResult:
    def __init__(self, matrix, sum, side, success) -> None:
        self.matrix = matrix
        self.sum = sum
        self.side = side
        self.success = success
